### ISSUE TITLE
Practice on mobile devices

### ISSUE THREAD

**Issue by tkainrad** [tkainrad/keycombiner#4]
Reported by users via mail.

**Is your feature request related to a problem? Please describe.**
KeyCombiner's interactive trainer is not displayed on devices that are too small, primarily smartphones. This means there is no way to practice on a smartphone, even if it is a text snippet collection that could, in theory, be practiced with a touchscreen keyboard. 

**Possible solutions**
I have these alternative ideas:
- Do not allow to practice keyboard shortcuts, due to the lack of a physical keyboard, but enable text snippet practice via a slightly modified trainer that fits on a small display.
- Let users practice text snippets by typing them on a touchscreen keyboard **and** use Anki-like flashcards for keyboard shortcuts. This means that shortcuts don't have to be typed but still appear in practice. A user can _flip the card_ and select via buttons whether they would have known the shortcut. Maybe there should be an option to use the same approach also for text snippets, as typing them on a touchscreen could be too tedious.

**Open questions**
How to gather statistics from these very different practice modes? How should this data then influence confidence values and such?

**Next Steps**
This will probably be done in iterations. In the first iteration, the results of such a mobile practice run will likely not be saved at all.

**Comment by QueensDeeJay** [comment]
For a solution, I vote for "...a slightly modified trainer" that would **allow text snippet practice on mobile devices**, such as smartphones and iPads.  

For me, that would be perfect; even if it meant only allowing snippet collections **_to be built_** on devices with physical keyboards (i.e. computers or iPad accessories). 

As I see it, this could be a two-step, productivity-boosting, workflow: 
Step 1) Input snippets at a desk, via keyboard
Step 2) Practice snippets on a mobile device, anywhere (in a line, a doctor's waiting room, on the subway, a train, etc.) 

Given virtually unlimited opportunities to practice, this platform could become invaluable, since the more snippets you know (and can use without a second thought) the more stress-free time is saved.

My use case: Electronic Shorthand. A great tool for journalists, clergy, writers, teachers, students, business people, parents... anyone who wants to be able to completely capture the moment at hand or the thoughts in mind.

**Comment by tkainrad** [comment]
Thank you for this additional input!

Your use case should definitely be possible. We recently added a _Practice Settings_ section that will work nicely for customizing the mobile practice experience.

Typing them in will most likely be the default for text snippets. However, with keyboard shortcuts, we will need some workaround (e.g., flashcards to flip) for lack of a physical keyboard with modifier keys.

> Given virtually unlimited opportunities to practice, this platform could become invaluable

I sure hope so :grinning:

**Comment by mikegomezg** [comment]
For "practicing" on a mobile device, I think just having flashcards with the command on one side and the combination on the other would be enough. You don't haven't to enter the combination to proceed, just signal whether or not you knew it.

**Comment by tkainrad** [comment]
> For "practicing" on a mobile device, I think just having flashcards with the command on one side and the combination on the other would be enough. You don't haven't to enter the combination to proceed, just signal whether or not you knew it.

Yes, that's what I had in mind, too. The main reason why I haven't implemented it yet is that the statistics for this mode would be incompatible with current practice statistics.
